src/Google/GlobalSiteTag.php

```diff
@@ -66,18 +67,57 @@ public function register(): void {
 
 		add_action(
 			'wp_head',
-			function() use ( $ads_conversion_id ) {
+			function () use ( $ads_conversion_id ) {
 				$this->activate_global_site_tag( $ads_conversion_id );
 			},
 			999998
 		);
 		add_action(
 			'wp_head',
-			function() use ( $ads_conversion_id, $ads_conversion_label ) {
+			function () use ( $ads_conversion_id, $ads_conversion_label ) {
 				$this->maybe_display_event_snippet( $ads_conversion_id, $ads_conversion_label );
 			},
 			1000000
 		);
+
+		add_action(
+			'wp_body_open',
+			function () {
+				$this->display_page_view_event_snippet();
+			},
+			1000001
+		);
+
+		add_action(
+			'wp_body_open',
+			function () {
+				$this->display_view_item_event_snippet();
+			},
+			1000002
+		);
+		add_action(
+			'wp_body_open',
+			function () {
+				$this->display_cart_page_snippet();
+			},
+			1000004
+		);
+		add_action(
+			'wp_body_open',
+			function () {
+				$this->display_purchase_page_snippet();
+			},
+			1000005
+		);
+
+		add_filter(
+			'wc_add_to_cart_message_html',
```

Comment:
I'm not sure if this is the most reliable way to track adding an item to the cart. Our other tracking solutions detect this quite differently. This is also not compatible with the WooCommerce all product / cart blocks. I think we need to do some deeper investigation in how to detect this reliably.

Reply:
Updated to "woocommerce_add_to_cart_button_html". Not sure if this is compatible with the Woocommerce all product/cart product. If it is not, could I get some of your suggestions? Thanks.

Reply:
That makes sense. Could I keep using ''wc_add_to_cart_message_html" here and add "TODO" for further investigation?

Reply:
I think that would be the best approach for now.